resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/MediaTypeMap.java

```diff
@@ -272,7 +272,7 @@ private CachedMediaTypeAndClass(Class clazz, MediaType mediaType)
       {
          this.clazz = new WeakReference(clazz);
          this.mediaType = mediaType;
-         int result = getClazz().hashCode();
+         int result = clazz.hashCode();

```

Comment:
In `clazz.hashCode()` `clazz `refers to the 1st argument of type `Class` and not to `this.clazz` of type `WeakReference<Class<?>>`. Your concern was maybe because of this ?